### DISCUSSION TITLE
Changing directories with fd

### DISCUSSION THREAD

**Discussion by weiying-chen** [sharkdp/fd#1495] [Q&A]
I found out I can change directories with fd like this:

```
cd $(fd directory)
```

Is this the only way? Or there are other alternatives?

**Comment by tmccombs** [comment]
Basically. Changing directories is something thay has to be done by the shell, so it can't be done directly by fd. You need fd to output the result and capture it then pass it to cd. What you have is the somplest way to do that.

**Comment by tavianator** [comment]
You might be interested in fuzzy-finders like https://github.com/junegunn/fzf

**Comment by kranurag7** [comment]
Hey there, 
I was going to GitHub discussion and found this discussion. I want to achieve something similar. 

I'm using `kustomize` and it only takes `present working directory` as input. 

Below is all the `kustomization.yaml` files and I want to run `kustomize edit fix --vars` by entering into all those directory and running the command `kustomize edit fix --vars` 

Manually, I can do it something like this: `cd config/certmanager` and `kustomize edit fix --vars`. After this I've to come back to root directory and go into other directory and run the same command as above. 

If `kustomize` would take any directory as input then I could have achieve it using `fd kustomization.yaml --exec kustomize edit fix {//}` and that would have worked but please note that this command requires you to run in that directory where `kustomization.yaml` is present meaning I'll have to `cd` into it. 

```bash
$ fd kustomization.yaml
config/certmanager/kustomization.yaml
config/crd/kustomization.yaml
config/default/kustomization.yaml
config/manager/kustomization.yaml
config/rbac/kustomization.yaml
config/webhook/kustomization.yaml
hack/observability/kustomization.yaml
```

Here's what I've tried but I was not able to achieve it. 

```bash
fd kustomization.yaml --exec-batch 'cd {//}; kustomize edit fix --vars {//}'
```

My thought process behind this was that it will `cd` into each directory one by one and run the command. Is my thinking correct here? Is it possible to achieve this using `fd`?

**Reply by tmccombs** [reply]
I think you would need to do something like 
```
fd kustomization.yaml --exec bash -c 'cd {//}; kustomize edit fix --vars'
```

**Reply by kranurag7** [reply]
Thank you very much, this worked for me.
I think this way, we don't have to worry about coming back to the parent directory once again because each exec is done in a new sub-shell.
